accepted/2022/dotnet-release-labels.md

```diff
@@ -0,0 +1,245 @@
+
+The industry examples demonstrate that we are using the "LTS" label correctly and the "Current" label incorrectly. The Ubuntu/Canonical model is the closest to our practice.
+
+## Proposed labels
+
+New labels are proposed for a variety of purposes.
+
+**Release-type (support duration) labels:**
+
+- **STS** -- Indicates a release is supported for the Short Term Support (STS) timeframe (18 months) and currently actively supported.
+- **LTS** -- Indicates a release is supported for the Long Term Support (LTS) timeframe (3 years) and currently actively supported.
+
+**Release support status labels:**
+
+- **Preview** -- Indicates a release is in preview and (unless "Go-Live") is unsupported.
+- **Go-Live** -- Indicates a preview or release candidate is supported in production.
+- **Maintenance** -- Indicates that a release is within 6 months of remaining support, with security fixes only.
+- **EOL** -- Indicates that the support has ended ("end of life").
+
+**Other proposed labels:**
+
+- **Latest** -- Indicates that a release is the latest actively supported .NET release (not a Preview or Release Candidate).
+- **TILL202212** -- Indicates that a release is supported until a particular month.
+
+Note: "Other proposed labels" are labels that were proposed and that may or may not get used. They are good ideas, so they are being documented. Various pages and viewers of support information can use these labels even if Microsoft chooses not to.
+
+Notes:
+ - The **Current** label will no longer be used and will be replaced with **STS**.
+ - Labels can use whatever casing makes sense for the artifact. For example, `releases.json` will use all lower-case.
+
+In some views, we can only show one label. In those cases, we will show the most relevant label, which will use the following algorithm:
+
+- show support status label, if
+  - support status is "Preview"
+  - support status is "Go Live"
+  - support status is (Maintenance or  EOL)
+- else show release type
+
+For example, the `support-phase` property in `releases-index.json` and `releases.json` will apply this algorithm.
+
+These labels will show up in multiple places, including:
+
+- [.NET Download pages](https://dotnet.microsoft.com/download/dotnet)
+- [Release notes](https://github.com/dotnet/core/blob/main/releases.md)
+- [releases.json](https://github.com/dotnet/core/blob/main/release-notes/releases-index.json)
+- Visual Studio
+
+## In practice
+
+Given this proposal, we will see the following labels used in the following timeframes (with a subset of [releases](https://github.com/dotnet/core/blob/main/releases.md) used as examples).
+
+
+Now (May 2022):
+
+- .NET Core 3.1 (LTS)
+- .NET 5 (EOL)
+- .NET 6 (LTS, Latest)
+- .NET 7 (Preview, STS)
+
+Just before .NET 7 GA (October, 2022):
+
+- .NET Core 3.1 (Maintenance; LTS)
+- .NET 5 (EOL)
+- .NET 6 (LTS, Latest)
+- .NET 7 (Go-Live, Preview, STS)
```

Comment:
seems this labelling is confusing to me especially in this example...
Go-Live - _supported in production_
Preview - _not supported in production, but because Go-Live supported in production_
STS - _(18 months) and currently actively supported_

I'm more along the lines of only having one current label...

**Odd** year releases
Preview --> Go-Live --> **LTS** --> Maintenance -->EOL
**Even** year releases
Preview --> Go-Live --> **STS** --> Maintenance -->EOL

maybe STS-Preview/LTS-Preview and STS-Go-Live/LTS-Go-Live would be slightly useful but STS-Maintenance/LTS-Maintenance (both mean security only) or _STS-_EOL/_LTS-_EOL (both mean dead) does not really add any value.

IMHO I dislike Latest/Current/TILLyyyyMM from how much confusion was caused by the docker tag "latest"

Reply:
There's really no value to having a single label. Also, compound labels are to avoided since they are objectively error prone and have bad performance characteristics. They require significant string handling beyond equality checks.